src/defrag.c

```diff
@@ -38,6 +38,10 @@
 
 #ifdef HAVE_DEFRAG
 
+#define defraged_alloc defrag_jemalloc_alloc
```

Comment:
what in case of no JEMALLOC compilation?

Reply:
Oh then we do not enable defrag... (forgot about it) still seem like this would have been better be addressed explicitly

Reply:
Ideally I wanted to have "pluggable" defrag interface, where  default defrag_jemalloc is one of the implementation, but one can plug different logic. Under this I wanted to initially remove the specific functions from defrag logic

Reply:
yea, we can add it when/if needed. will fix

Reply:
Idea: The logic for the different allocators can be encapsulated in `allocator_defrag.c`, which should just expose functions named `defraged_alloc`, etc. without "jemalloc" in the names.

We can conditionally define the `defraged_alloc` function for jemalloc and potentially for another allocator's internals in the future, if we ever add defrag support for another one.

Reply:
@zuiderkwast this was exactly the intent, agree about the naming, will change

Reply:
resolving comments to indicate they where fixed in the new [PR](https://github.com/valkey-io/valkey/pull/1266)